packages/deployments/contracts/contracts/core/connext/helpers/Executor.sol

```diff
@@ -227,6 +227,19 @@ contract Executor is IExecutor {
       _sendToRecovery(isNative, hasValue, _args.assetId, payable(_args.to), payable(_args.recovery), _args.amount);
     }
 
+    // If there is unclaimed tokens, send them to the recovery address
+    if (hasValue) {
+      uint256 unclaimed = _args.amount - (beforeBalance - _assetBalance(_args.assetId));
+      if (unclaimed > 0) {
+        _sendToRecovery(isNative, hasValue, _args.assetId, payable(_args.to), payable(_args.recovery), unclaimed);
+      }
+    }
+
+    // Reset allowance to 0!
+    if (!isNative && hasValue) {
+      SafeERC20.safeApprove(IERC20(_args.assetId), _args.to, 0);
```

Comment:
We should only have to reset allowance to 0 if there were "unclaimed" / unspent tokens, right?

If all the tokens were spent, then wouldn't the allowance now be 0?